### ISSUE TITLE
Random button selection should be resolved after game acceptance

### ISSUE THREAD

**Issue by blackshadowshade** [buttonmen-dev/buttonmen#1750]
Currently, when a game is created including random button selection and the opponent doesn't have auto-accept, the opponent can see which buttons have been selected before the game acceptance decision has been made.

Random buttons shouldn't resolve until after the game has been accepted.

**Comment by irilyth** [comment]
Hmm, I think not everyone would agree with this: Some folks have said that they want the option to reject a game if they don't like their random button.

Maybe this should be an option at game-creation time?

**Comment by blackshadowshade** [comment]
In this case, perhaps two types of Random Button:
- Random Button (visible)
- Random Button (hidden)

**Comment by irilyth** [comment]
Sure, that'd work, although people would have to know what "hidden" and "visible" meant here; whereas if it's a checkbox, it could have more text, like "Reveal random button(s) before opponent accepts game:" or something.

**Comment by AdmiralJota** [comment]
I agree that the current behavior definitely seems like a bug. I could see having an option for a thing like this at some point in the future, but IMO the default should definitely be to resolve the buttons after both players have joined.

At this point, I don't recall how the open games were implemented; was there a reason that was handled differently from how they work?

**Comment by irilyth** [comment]
The argument for the default being "Reveal random button(s) before opponent accepts game" is that people have explicitly said that they want the ability to reject games that include buttons they don't like, and random-button games are a big source of those.

I think the reject-game functionality will disappoint a significant chunk of its target audience if reveal-before-accept isn't the default.

**Comment by AdmiralJota** [comment]
Wouldn't that make the ability to reject games all the [i]more[/i] useful?

And in any case, players will always have the option to create games where the opponent's button [i]isn't[/i] random, and soon, they'll even be able to let the opponent choose their own button.

To me, it feels like it rather defeats the whole point of random button selection if you already know what the "random" button is... it'd be like only accepting games where you already knew what your initial die rolls were going to be.

**Comment by irilyth** [comment]
> Wouldn't that make the ability to reject games all the [i]more[/i] useful?

I'm not sure which thing "that" is here.

I guess it depends on what you think the point of random button selection is. :^) For me, it's more like "I don't want to have to choose buttons" than like "I want to commit to playing a game no matter what the button is". Getting to mulligan your initial die roll would be a game-altering rules change. Getting to veto a random button is more like reaching into your bag of buttons and pulling one out, saying "nah, I don't feel like playing Bennett again right now", and throwing it back and pulling out another one. Unless the *meta* game rules require you to play whatever button you get, I think it's fine to say "a random button, but not if I get one that's no fun".

Ah, but that's for sure interesting: In *tournaments* where the organizer says buttons are chosen randomly, you definitely don't want people vetoing their random button, because that's a great example of the "there is a meta game" thing.

Anyway, point is: For my personal enjoyment of the game, in general for a series of for-fun random vs random games against other people, I'd much rather let people toss back their random buttons and pick another one than force people to play a button the don't want.

**Comment by dwvanstone** [comment]
I think most of the people who wanted to reject Random buttons wanted to reject them because they weren't Tournament Legal. Maybe down the line we could add TL-Random or some other option that restricts random to TL. 

But for now, if I create a random v. random game I would expect the person accepting to not get to see what the buttons are. Otherwise it's unfair for me -- I create 10 random v. random games, and they get to accept just the ones that are to their advantage. 

If they don't want to play random v. random games in general they can reject them all. This would be an improvement.

This bug should not block the November release, in my opinion.

**Comment by irilyth** [comment]
> I create 10 random v. random games, and they get to accept just the ones that are to their advantage.

This is not my use case -- my use case is that you and I finish a game, I click "create another game with random buttons, preserving chat". In that case, I don't think it's "unfair" for you to say "this is a disadvantageous matchup, I don't wanna play it", if you feel that way. (Myself, I'd just as soon play it, and the reverse, and see what happens. :^) But if you don't like playing imbalanced games, I still want to be able to play random vs random games with you.)

When do you create 10 random v random games? Like against the same person?

Or are you talking about Open games?

**Comment by irilyth** [comment]
In any case, to recap, my argument is that there should be a checkbox, so that you the game creator can do it whichever way you want.

I'd make it default to "reveal", but I guess I don't feel super strongly about that, although I'll personally have to click on this a lot if the default is "hide". :^) In which case I can argue for a user preference, but whatever.

If it'd easy to make it default to "hide" for *open* games, I think that'd be very sensible.

I agree that this isn't for November. Is it for December?

**Comment by AdmiralJota** [comment]
As a side note regarding which to make the default: the old site [i]only[/i] had hidden random. I think that this thread is the first time I've heard of anyone wanting non-hidden random. So while I'm certainly not opposed to it being an option...

**Comment by cgolubi1** [comment]
So, i don't want to get too philosophical about this decision-making process, but one thing that occurred to me is that the last couple of months of development and design have been so contentious because the work we've been doing has been changing defaults --- this means trading off one player's use cases against another player's use cases, and in particular breaking some use cases that work right now in order to enable new use cases.  Adding new functionality, by contrast, is less contentious, because players/testers who don't think they value the new functionality can just ignore it --- it doesn't directly hurt them for it to exist. 

So i think it's fine to speculate about this, but it's a sort of terrible priority given the implementations which are being discussed, because we're having yet another argument about whose use cases are important and whose use cases should be sacrificed by being caused to no longer work.  My alternate suggestions:
* In the button choice pulldown, make two options for "Random Button" --- "Random Button (reveal immediately)" and "Random Button (hide until game is accepted)".  This is the analogue to irilyth's checkbox, but it doesn't require a new UI element.
* Ignore this issue until we work on #1234.  The old site had things like #1234 in spades, and i'd argue that it's much more important to give players that new functionality than to change defaults around.

My meta-suggestion is that we do not consider this a December priority unless we get player feedback that the reveal-immediately behavior is hurting people, but leave it in the queue for a future month.

**Comment by dwvanstone** [comment]
Agreed that this is not a high priority item unless we get player feedback saying it should be. It's an interesting example of mixtures of use cases that I'm still mulling over and will enjoy talking about it with people offline.
